xarray/core/variable.py

```diff
@@ -302,6 +302,40 @@ def data(self, data):
                 "replacement data must match the Variable's shape")
         self._data = data
 
+    def astype(self, dtype, casting = 'unsafe', copy = True):
+        """
+        Copy of the Variable object, with data cast to a specified type.
+
+        Parameters
+        ----------
+        dtype : str or dtype
+            Typecode or data-type to which the array is cast.
+        casting : {'no', 'equiv', 'safe', 'same_kind', 'unsafe'}, optional
+            Controls what kind of data casting may occur. Defaults to 'unsafe'
+            for backwards compatibility.
+
+            * 'no' means the data types should not be cast at all.
+            * 'equiv' means only byte-order changes are allowed.
+            * 'safe' means only casts which can preserve values are allowed.
+            * 'same_kind' means only safe casts or casts within a kind,
+                like float64 to float32, are allowed.
+            * 'unsafe' means any data conversions may be done.
+        copy : bool, optional
+            By default, astype always returns a newly allocated array. If this
+            is set to False and the `dtype` requirement is satisfied, the input
+            array is returned instead of a copy.
+
+        See also
+        --------
+        np.ndarray.astype
+        dask.array.Array.astype
+        """
+        self.data = duck_array_ops.astype(self.data,
+                                          dtype,
+                                          casting = casting,
+                                          copy = copy)
+        return self
```

Comment:
This should return a new `Variable` object rather than modifying this one in place. Take a look at the methods below for examples. This could be something like:
```python
data = self.data.astype(dtype, casting=casting, copy=copy)
return type(self)(dims, data, self._attrs, self._encoding, fastpath=True)
```